Sources/Gravatar/Network/Services/URLSessionHTTPClient.swift

```diff
@@ -62,11 +62,21 @@ extension HTTPClientError {
     func map() -> ResponseErrorReason {
         switch self {
         case .URLSessionError(let error):
-            .URLSessionError(error: error)
+            return .URLSessionError(error: error)
         case .invalidHTTPStatusCodeError(let response, let data):
-            .invalidHTTPStatusCode(response: response, data: data)
+            if response.statusCode == 400 {
+                do {
+                    let error: ModelError = try data.decode()
+                    return .invalidHTTPStatusCode(response: response, errorPayload: error)
+                } catch {
+                    // if parsing the error fails then return invalidHTTPStatusCode without the payload.
+                    return .invalidHTTPStatusCode(response: response)
+                }
```

Comment:
Since we are skipping handling the parsing error, maybe we can avoid using try-catch?

```suggestion
                let error: ModelError? = try? data.decode()
                return .invalidHTTPStatusCode(response: response, errorPayload: error)
```

Reply:
Nice. 👍